### DISCUSSION TITLE
Unique metricName for AWS CloudWatch scaler based on labels or other identifiers

### DISCUSSION THREAD

**Discussion by ivankovnatsky** [kedacore/keda#2931] [General] [answered]
Hello, followed this page: https://keda.sh/docs/2.6/scalers/aws-cloudwatch/ to configure KEDA to use AWS CloudWatch to scale based on Kafka topic lags.

I wonder if there's any way to set metricName to be completely unique, like custom-custom. Would be just simpler to read the values like in here:

```
k get scaledobjects.keda.sh -A --no-headers |awk '{print $2}'|while read name; do kubectl get --raw "/apis/external.metrics.k8s.io/v1beta1/namespaces/blues/s0-aws-cloudwatch-Topic?labelSelector=scaledobject.keda.sh%2Fname%3D${name}"|jq;done

{
  "kind": "ExternalMetricValueList",
  "apiVersion": "external.metrics.k8s.io/v1beta1",
  "metadata": {},
  "items": [
    {
      "metricName": "s0-aws-cloudwatch-Topic",
      "metricLabels": null,
      "timestamp": "2022-04-23T12:49:26Z",
      "value": "0"
    }
  ]
}
{
  "kind": "ExternalMetricValueList",
  "apiVersion": "external.metrics.k8s.io/v1beta1",
  "metadata": {},
  "items": [
    {
      "metricName": "s0-aws-cloudwatch-Topic",
      "metricLabels": null,
      "timestamp": "2022-04-23T12:49:26Z",
      "value": "0"
    }
  ]
}
```

I've noticed that for some scalers we can set `metricName`, like here: https://github.com/kedacore/keda-docs/blob/28aae9ea708a0dab9cc52856b60a75aaeb977073/content/docs/2.5/scalers/azure-storage-blob.md. But for AWS CloudWatch that is probably not the case.

Any way to do it? If not the name, maybe I can play with it using labels, so that it would show under `"metricLabels"` 

Thanks,
Ivan.

**Comment by zroubalik** [comment, marked as answer]
Hi, you can use this approach: https://keda.sh/docs/2.6/operate/metrics-server/#how-to-get-metric-when-multiple-scaledobjects-have-the-same-metric-name

**Reply by ivankovnatsky** [reply]
Yeah, thanks, I was just wondering if can name the metric name somehow. It appears I can't.

**Reply by zroubalik** [reply]
We are trying to reduce this, as the metric name is just an implementation detail of KEDA internals. Is there anything specific you need it for?

**Reply by ivankovnatsky** [reply]
Basically just for the purpose of representability, an improvement as an identifier to the output like here:

```
{
  "kind": "ExternalMetricValueList",
  "apiVersion": "external.metrics.k8s.io/v1beta1",
  "metadata": {},
  "items": [
    {
      "metricName": "s0-aws-cloudwatch-Topic-unique-identifier-1",
      "metricLabels": null,
      "timestamp": "2022-04-23T12:49:26Z",
      "value": "0"
    }
  ]
}
{
  "kind": "ExternalMetricValueList",
  "apiVersion": "external.metrics.k8s.io/v1beta1",
  "metadata": {},
  "items": [
    {
      "metricName": "s0-aws-cloudwatch-Topic-unique-identifier-1",
      "metricLabels": null,
      "timestamp": "2022-04-23T12:49:26Z",
      "value": "0"
    }
  ]
}
```

But, fairly speaking that is not very-very important, just a matter of OCD perhaps.

**Reply by zroubalik** [reply]
I see the point, we just try to make the UI (trigger spec) simple as possible to avoid confusion, that users might have. It's just about finding the right balance, as is this is internal thing, I am not inclined towards adding this field.

**Reply by ivankovnatsky** [reply]
I do agree. Anyway, we do really excited about the KEDA, thanks for your work.
